developer/js/source/package-compiler/kmp-compiler.ts

```diff
@@ -145,8 +145,8 @@ export default class KmpCompiler {
     zip.file(kmpJsonFileName, JSON.stringify(kmpJsonData, null, 2));
 
     // Generate kmp file
-    var data = zip.generate({base64:false,compression:'DEFLATE'});
-    fs.writeFileSync(kmpFileName, data, 'binary');
-
+    zip.generateAsync({type: 'binarystring', compression:'DEFLATE'}).then(function (data) {
+      fs.writeFileSync(kmpFileName, data, 'binary');
+    });
```

Comment:
This is no longer synchronous, so I am concerned about side-effects. I'd like to investigate this further before we went ahead and merged.

Reply:
At the very least, `buildKmpFile` needs to return the `Promise` from `zip.generateAsync.then`, and the caller needs to handle it, to ensure that the zip file is actually ready for use before the next step in the compile. This potentially breaks a lot of the design, which has assumed synchronous use.

Reply:
I've updated `buildKmpFile` to return the Promise, which also involved moving the call to `writeFileSync`.